api/auth/client.go

```diff
@@ -18,171 +18,157 @@ type Auth struct {
-
-	_, err := store.api.
-		URL("/auth/api/v1/idp/clients/%s", idpID).
+// DeleteIdpClient delete identity provider client configuration by Id.
+func (c *Auth) DeleteIdpClient(idpId string) error {
+	_, err := c.api.
+		URL("/auth/api/v1/idp/clients/%s", url.PathEscape(idpId)).
 		Delete()
 
 	return err
 }
 
 // RegenerateIdpClientConfig regenerates client_id and client_secret
 // for OIDC identity provider client configuration.
-func (store *Auth) RegenerateIdpClientConfig(idpID string) (*IdpClientConfig, error) {
+func (c *Auth) RegenerateIdpClientConfig(idpId string) (*IdpClientConfig, error) {
 	clientConfig := &IdpClientConfig{}
 
-	_, err := store.api.
-		URL("/auth/api/v1/idp/clients/%s/regenerate", url.PathEscape(idpID)).
+	_, err := c.api.
+		URL("/auth/api/v1/idp/clients/%s/regenerate", url.PathEscape(idpId)).
 		Post(nil, &clientConfig)
 
 	return clientConfig, err
 }
 
-// UserSessions fetches valid sessions by userID.
-func (store *Auth) UserSessions(offset, limit int, sortkey, sortdir, userID string) (*sessionsResult, error) {
-	filters := Params{
-		Offset:  offset,
-		Limit:   limit,
-		Sortkey: sortkey,
-		Sortdir: sortdir,
-	}
-	userSessions := &sessionsResult{}
+// MARK: Session Storage
+// GetUserSessions get valid sessions by userID.
+func (c *Auth) GetUserSessions(filters *response.FilterParams, userId string) (*response.Page[Session], error) {
+	userSessions := &response.Page[Session]{}
 
-	_, err := store.api.
-		URL("/auth/api/v1/sessionstorage/users/%s/sessions", userID).
+	_, err := c.api.
+		URL("/auth/api/v1/sessionstorage/users/%s/sessions", url.PathEscape(userId)).
 		Query(&filters).
 		Get(&userSessions)
 
 	return userSessions, err
 }
 
-// SourceSessions fetches valid sessions by sourceID.
-func (store *Auth) SourceSessions(offset, limit int, sortkey, sortdir, sourceID string) (*sessionsResult, error) {
-	filters := Params{
-		Offset:  offset,
-		Limit:   limit,
-		Sortkey: sortkey,
-		Sortdir: sortdir,
-	}
-	sourceSessions := &sessionsResult{}
+// GetSourceSessions get valid sessions by sourceID.
+func (c *Auth) GetSourceSessions(filters *response.FilterParams, sourceId string) (*response.Page[Session], error) {
+	sourceSessions := &response.Page[Session]{}
 
-	_, err := store.api.
-		URL("/auth/api/v1/sessionstorage/sources/%s/sessions", sourceID).
+	_, err := c.api.
+		URL("/auth/api/v1/sessionstorage/sources/%s/sessions", url.PathEscape(sourceId)).
 		Query(&filters).
 		Get(&sourceSessions)
 
 	return sourceSessions, err
 }
 
 // SearchSessions searches for sessions
-func (store *Auth) SearchSessions(offset, limit int, sortkey, sortdir string, search *SearchParams) (*sessionsResult, error) {
-	filters := Params{
-		Offset:  offset,
-		Limit:   limit,
-		Sortkey: sortkey,
-		Sortdir: sortdir,
-	}
-	sessions := &sessionsResult{}
+func (c *Auth) SearchSessions(filters *response.FilterParams, search *SessionSearchRequest) (*response.Page[Session], error) {
```

Comment:
Probably no reason to pass pointer. As FilterParams go through JSON marshalling before being url-encoded the "omitempty" tag would mean all empty fields are dropped. In other words, an empty FilterParams can substitute nil.